packages/realm/src/Results.ts

```diff
@@ -104,6 +114,48 @@ export class Results<T = unknown> extends OrderedCollection<T> {
     }
   }
 
+  /**
+   * Add this query result to the set of active subscriptions. The query will be joined
+   * via an `OR` operator with any existing queries for the same type.
+   *
+   * @param options Options to use when adding this subscription (e.g. a name or wait behavior).
+   * @returns A promise that resolves to this {@link Results} instance.
+   * @experimental This API is experimental and may change or be removed.
+   */
+  async subscribe(options: SubscriptionOptions = { behavior: WaitForSync.FirstTime }): Promise<this> {
+    const subs = this.realm.subscriptions;
+    const shouldWait =
+      (options.behavior === WaitForSync.FirstTime && !subs.exists(this)) || options.behavior === WaitForSync.Always;
+    if (shouldWait) {
+      if (typeof options.timeout === "number") {
+        await new TimeoutPromise(
+          subs.update((mutableSubs) => mutableSubs.add(this, options)),
+          { ms: options.timeout, rejectOnTimeout: false },
+        );
+      } else {
+        await subs.update((mutableSubs) => mutableSubs.add(this, options));
+      }
+    } else {
+      subs.updateNoWait((mutableSubs) => mutableSubs.add(this, options));
+    }
+
+    return this;
+  }
+
+  /**
+   * Unsubscribe from this query result.
+   * @experimental This API is experimental and may change or be removed.
+   */
+  unsubscribe(): void {
```

Comment:
Should we specify that this is non-blocking?

Reply:
I think the non-blocking part is already documented via the `void` (rather than `Promise<void>`) return type. But it wouldn't hurt to add to the TS docs that it will not wait for the update to complete, I'll add that 🙂

Reply:
I've added some clarification to the TS docs ✅